django/urls/converters.py

```diff
@@ -0,0 +1,91 @@
+import uuid
+
+from django.utils import lru_cache
+
+
+class BaseConverter(object):
+    pass
+
+
+class IntConverter(BaseConverter):
+    regex = '[0-9]+'
+
+    def to_python(self, value):
+        return int(value)
+
+    def to_url(self, value):
+        return str(value)
+
+
+class StringConverter(BaseConverter):
+    regex = '[^/]+'
+
+    def to_python(self, value):
+        return value
+
+    def to_url(self, value):
+        return value
+
+
+class UUIDConverter(BaseConverter):
+    regex = '[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}'
+
+    def to_python(self, value):
+        return uuid.UUID(value)
+
+    def to_url(self, value):
+        return str(value)
+
+
+class SlugConverter(BaseConverter):
+
+    # Django has both `slug_re` and `slug_unicode_re`. In the case of URLs, the
+    # non-unicode variant seems to make more sense.
+    regex = '[-a-zA-Z0-9_]+'
+
+    def to_python(self, value):
+        return value
+
+    def to_url(self, value):
+        return value
+
+
+class PathConverter(BaseConverter):
```

Comment:
Since this converter matches anything, I'm wondering if "path converter" is the best name.

Perhaps `BaseConverter` should be implemented like this and registered under the "any" name (or similar).

What do you think?

Reply:
Reading below, I see that Flask has an "any" converter that does something more complicated. Creating a converter with the same name but a different behavior doesn't sound good.

Reply:
Let's not block the merge on this. The current implementation matches the DEP which was largely discussed.

I'm brainstorming to avoid future problems. Seeing the code sometimes gives new ideas.